src/structures/paging/page.rs

```diff
@@ -279,7 +279,7 @@ impl<S: PageSize> Sub<Self> for Page<S> {
 }
 
 /// A range of pages with exclusive upper bound.
-#[derive(Clone, Copy, PartialEq, Eq)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
Same point as above: I don't think that deriving `PartialOrd`/`Ord` makes sense for range types.